src/main/scala/edu/ie3/simona/io/result/ResultEntityCsvSource.scala

```diff
@@ -74,16 +75,18 @@ final case class ResultEntityCsvSource(
   ): Future[_ <: ResultEntity] = Future {
 
     val simpleEntityData = new SimpleEntityData(rowWithHeadline.asJava, model)
-    resultFactory.get(simpleEntityData).toScala match {
-      case Some(resultEntity) =>
-        resultEntity
-      case None =>
-        logger.error(
-          s"Unable to parse line in $filePath\nline: ${rowWithHeadline.values}\nWill ignore that line!"
-        )
-        throw new ProcessResultEventException(
-          s"Unable to parse line in $filePath\nline: ${rowWithHeadline.values}\nWill ignore that line!"
-        )
+    val option: Try[SystemParticipantResult, FactoryException] =
+      resultFactory.get(simpleEntityData)
+
+    if (option.isSuccess) {
+      option.getData.get()
+    } else {
+      logger.error(
+        s"Unable to parse line in $filePath\nline: ${rowWithHeadline.values}\nWill ignore that line!"
+      )
+      throw new ProcessResultEventException(
+        s"Unable to parse line in $filePath\nline: ${rowWithHeadline.values}\nWill ignore that line!"
+      )
```

Comment:
Three things:

1. This class probably can be removed since it is not used anywhere. And if it would be needed we want that kind of functionality within the PSDM
2. Logging an error and throwing an exception seems unneccessary an exception is sufficient
3. If you get an exception from that try (which is not an option so it should not be called option) you would want to at least append the exception to the exception you are throwing otherwise the information is lost

Reply:
In that case you can double check if any of the code is used and if not just delete the file

Reply:
Should I add this functionality to the PSDM?